plugins/UM3NetworkPrinting/resources/qml/PrinterCard.qml

```diff
@@ -0,0 +1,236 @@
+                            anchors.fill: parent;
+                            color: UM.Theme.getColor("monitor_tab_text_inactive");
+                            elide: Text.ElideRight;
+                            font: UM.Theme.getFont("default");
+                            text: {
+                                if (printer.state == "disabled") {
+                                    return catalog.i18nc("@label", "Not available");
+                                } else if (printer.state == "unreachable") {
+                                    return catalog.i18nc("@label", "Unreachable");
+                                }
+                                if (printer.activePrintJob != null) {
+                                    return printer.activePrintJob.name;
+                                }
+                                return catalog.i18nc("@label", "Available");
+                            }
+                            visible: printer;
+                        }
+                    }
+                }
+
+                // Collapse icon
+                UM.RecolorImage {
+                    id: collapseIcon;
+                    anchors {
+                        right: parent.right;
+                        rightMargin: UM.Theme.getSize("default_margin").width;
+                        verticalCenter: parent.verticalCenter;
+                    }
+                    color: UM.Theme.getColor("text");
+                    height: 15; // TODO: Theme!
+                    source: root.collapsed ? UM.Theme.getIcon("arrow_left") : UM.Theme.getIcon("arrow_bottom");
+                    sourceSize.height: height;
+                    sourceSize.width: width;
+                    visible: printer;
+                    width: 15; // TODO: Theme!
+                }
+
+                MouseArea {
+                    anchors.fill: parent;
+                    enabled: printer;
+                    onClicked: {
+                        if (model && root.collapsed) {
+                            printerList.currentIndex = model.index;
+                        } else {
+                            printerList.currentIndex = -1;
+                        }
+                    }
+                }
+
+                Connections {
+                    target: printerList
+                    onCurrentIndexChanged: {
+                        if (!model) {
+                            return;
+                        }
+                        root.collapsed = printerList.currentIndex != model.index;
+                    }
+                }
+            }
+
+            // Detailed card
+            PrinterCardDetails {
+                collapsed: root.collapsed;
+                printer: root.printer;
+                visible: root.printer;
+            }
+
+            // Progress bar
+            PrinterCardProgressBar {
+                visible: printer && printer.activePrintJob != null && printer.activePrintJob != undefined;
```

Comment:
ProTip: A (weak) comparison with `null` catches both `null` and `undefined`.

Reply:
I know but I'm not sure what you would do differently here.

Checking just for `printer.activePrintJob` throws warnings in the log (and lots of them) if `printer` is undefined.

Reply:
What @sedwards2009 is saying is that you can drop the ``` && printer.activePrintJob != undefined```

Reply:
Oh yeah, true. Sorry, didn't even think about that thanks to GitHub's layout.